tls/s2n_server_extensions.c

```diff
@@ -175,6 +175,7 @@ int s2n_recv_server_alpn(struct s2n_connection *conn, struct s2n_stuffer *extens
 int s2n_recv_server_status_request(struct s2n_connection *conn, struct s2n_stuffer *extension)
 {
     conn->status_type = S2N_STATUS_REQUEST_OCSP;
+    conn->handshake.handshake_type |= OCSP_STATUS;
```

Comment:
I initially attempted that approach, but we s2n_conn_set_handshake_type before reading the extensions:
https://github.com/awslabs/s2n/blob/7978d5608662860f47cae7176ab45597841b5776/tls/s2n_server_hello.c#L75-L86

I went with the path of least resistance, but let me take another pass to see if I can come up with a clean way to move s2n_conn_set_handshake_type after we read the extensions.